src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -151,84 +151,57 @@ bool TryLoadScannableAssembly(string assemblyPath, AssemblyScannerResults result
             }
         }
 
-        void ScanAssemblyAndDependencies(Assembly assembly, Dictionary<string, bool> processed, AssemblyScannerResults results)
+        bool ScanAssemblyAndDependencies(Assembly assembly, Dictionary<string, bool> processed, AssemblyScannerResults results)
         {
             if (assembly == null)
             {
-                return;
+                return false;
             }
 
-            if (processed.ContainsKey(assembly.FullName))
+            if (processed.TryGetValue(assembly.FullName, out var value))
             {
-                return;
+                return value;
             }
 
-            if (!ShouldScanAssembly(assembly))
-            {
-                processed[assembly.FullName] = false;
-            }
+            processed[assembly.FullName] = false;
 
             if (assembly.GetName().Name == CoreAssemblyName)
             {
                 processed[assembly.FullName] = true;
-                AddTypesToResult(assembly, results);
-                return;
```

Comment:
this would cause the nsb.core references to be scanned although that's not needed?